File: 11/2.rb
require_relative "selection_sort"

array = ["tar", "tommy", "army", "loser", "mary", "rat"]

def sort_letters_in_words(array)
  array.each_with_object([]) do |word, arr|
    letters = word.split ""
    arr << selection_sort(letters).join("")
  end
end

def anagram_sort(array)
  array = sort_letters_in_words array
  hash = {}
  array.each do |word|
    if hash[word].nil?
      hash[word] = 1
    else
      hash[word] += 1
    end
  end
  hash.map { |key, value| Array.new(value) { key } }.flatten.sort
end

p anagram_sort array

File: 11/python/2.py
from selection_sort import selection_sort

words = ["aa", "art", "army", "tommy", "mary", "rat", "jim", "tar"]

def sort_anagrams(words):
    words = ["".join(selection_sort(list(word))) for word in words]
    return selection_sort(words)

print sort_anagrams(words)
